nautobot/extras/models/relationships.py

```diff
@@ -303,14 +301,19 @@ def required_related_objects_errors(
 
             if initial_data is not None:
 
-                supplied_data = []
+                missing_data = False
 
                 if output_for == "ui":
                     supplied_data = initial_data.get(field_key, [])
+                    if not supplied_data:
+                        missing_data = True
+
                 elif output_for == "api":
                     supplied_data = initial_data.get(relation, {}).get(opposite_side, {})
+                    if len(supplied_data) == 0:
```

Comment:
Any reason not to align the patterns used her to check there's any `supplied_data`?

```suggestion
                    if  not supplied_data:
```

Reply:
Ah! I think I was told off by a linter or even flake8 before because I wasn't being explicit enough, but maybe it wasn't setup correctly. So every time I had to check an int was 0, I just used `len(i) == 0` instead of just `not i`. Good to know  I can use the implicit check again now!